python/ambassador/diagnostics/envoy_stats.py

```diff
@@ -107,7 +107,15 @@ def cluster_stats(self, name: str) -> Dict[str, Union[str, bool]]:
         when = self.last_update
         cstat = self.clusters
 
-        if name not in cstat:
+        cluster_exists = False
+        cname = None
+        for cluster_name in cstat.keys():
```

Comment:
@alexeikmun - Like I mentioned offline, I see this works. However, I'm hesitant to go forward with this approach mostly due to the question of `Does it scale for large clusters?` 

The `clusters_stats` function gets called for every cluster which means then this loop logic gets called for every cluster. So would this scale on a cluster with thousands of clusters and not hang? 

Also, it feels like we are losing the benefits of `cstats` being a dictionary. The purpose of that data structure is to allow us to do quick key lookups vs linear searches.